pi4_software/srauv_main.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+#  srauv_main.py
+#  SRAUV main control program. Has an pdate loop that operates at a
+#    deterministic rate (target 60 hz, min 10 hz), to make decisions
+#    based on current vehicle state and sensor values and apply the
+#    appropriate thrust until the next update cycle.
+#
+#  Update Loop:
+#    update_telemetry_values()
+#    estimate_position()
+#    evaluate_state()
+#    calculate_thrust()
+#    log_state()
+#    apply_thrust()
+#      
+#  Threaded I/O operations that update values via shared memory:
+#    distance sensor, imu, internal socket messaging
+
+import json
+import sys
+import socket
+import time
+import threading
+import multiprocessing
+from datetime import datetime
+from time import perf_counter
+
+# Custome imports
+import distance_sensor
+import thruster_controller
+import timestamp
+import command_msg
+import telemetry_msg
+import internal_socket_server
+import logger
+from srauv_settings import SETTINGS
+
+###################  Globals  ###################
+
+UPDATE_INTERVAL_MS = SETTINGS["update_interval_ms"]
+TEL_TX_INTERVAL_MS = SETTINGS["tel_tx_interval_ms"]
+main_internal_address = (SETTINGS["internal_ip"],SETTINGS["main_msg_port"])
+source = "srauv_main"
+starting_state = "idle"
+can_thrust = False
+threads = [] # contains all threads
+dist_sensor_values = [0.0, 0.0, 0.0, 0.0, 0.0]
+ds_threads = []
+thrust_values = [0, 0, 0, 0, 0, 0] # int, -100 to 100 as percent max thrust
+th_threads = []
+internal_socket_threads = []
+
+cmd_msg = command_msg.make(source, "sim")
+tel_msg = telemetry_msg.make(source, "sim")
+
+log_filename = str(f'Logs/{datetime.now().strftime("SR--%m-%d-%Y_%H-%M-%S")}.log')
```

Comment:
Might make sense to let the logger handle the creation of the log file name and just pass in "SR", "IS", etc